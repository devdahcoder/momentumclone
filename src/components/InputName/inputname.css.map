{
    "version": 3,
    "mappings": "AAAA,AAAA,qBAAqB,CAAC;EAClB,UAAU,EAAE,MAAM;EAClB,WAAW,EAAE,MAAM;EACnB,WAAW,EAAE,MAAM;EACnB,MAAM,EAAE,cAAc;EACtB,OAAO,EAAE,WAAW;EACpB,QAAQ,EAAE,QAAQ;EAClB,SAAS,EAAE,IAAI;EACf,WAAW,EAAE,GAAG;CAoEnB;;AA5ED,AAYI,qBAZiB,CAYjB,KAAK,CAAC;EACF,KAAK,EAAE,KAAgB;EACvB,SAAS,EAAE,IAAI;EACf,UAAU,EAAE,GAAG;EACf,cAAc,EAAE,GAAG;EACnB,WAAW,EAAE,mKAWH;EACV,OAAO,EAAE,YAAY;EACrB,aAAa,EAAE,GAAG;EAClB,OAAO,EAAE,CAAC;EACV,QAAQ,EAAE,MAAM;EAChB,WAAW,EAAE,MAAM;EACnB,SAAS,EAAE,yBAAyB;EACpC,MAAM,EAAE,IAAI;EACZ,SAAS,EAAE,KAAK;EAChB,KAAK,EAAE,IAAI;EACX,WAAW,EAAE,GAAG;EAChB,SAAS,EAAE,IAAI;EACf,UAAU,EAAE,MAAM;EAClB,QAAQ,EAAE,QAAQ;EAClB,GAAG,EAAE,CAAC;EACN,KAAK,EAAE,CAAC;EACR,MAAM,EAAE,CAAC;EACT,IAAI,EAAE,CAAC;EACP,aAAa,EAAE,GAAG,CAAC,KAAK,CAAC,KAAgB;EACzC,yBAAyB,EAAE,CAAC;EAC5B,0BAA0B,EAAE,CAAC;EAC7B,MAAM,EAAE,IAAI;CACf;;AAED,UAAU,CAAV,KAAU;EACN,EAAE;IACE,UAAU,EAAE,wBAAuB;;EAGvC,IAAI;IACA,UAAU,EAAE,sBAAsB;;;;AA1D9C,AA8DI,qBA9DiB,CA8DjB,iBAAiB,CAAC;EACd,UAAU,EAAE,MAAM;EAClB,WAAW,EAAE,MAAM;EACnB,SAAS,EAAE,KAAK;EAChB,cAAc,EAAE,GAAG;EACnB,SAAS,EAAE,IAAI;EACf,OAAO,EAAE,YAAY;EACrB,aAAa,EAAE,IAAI;EACnB,OAAO,EAAE,CAAC;EACV,QAAQ,EAAE,MAAM;EAChB,WAAW,EAAE,MAAM;EACnB,UAAU,EAAE,OAAO;EACnB,SAAS,EAAE,IAAI;CAClB",
    "sources": [
        "inputname.scss"
    ],
    "names": [],
    "file": "inputname.css"
}