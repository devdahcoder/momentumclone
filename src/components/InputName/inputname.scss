.input-name-container {
    text-align: center;
    line-height: normal;
    white-space: nowrap;
    margin: -5px -2px -8px;
    padding: 5px 2px 8px;
    position: relative;
    font-size: 100%;
    font-weight: 400;



    input {
        color: hsl(0, 0%, 100%);
        font-size: 100%;
        background: 0 0;
        vertical-align: top;
        font-family: Inter,
        -apple-system,
        BlinkMacSystemFont,
        'Segoe UI',
        'Roboto',
        'Oxygen',
        'Ubuntu',
        'Cantarell',
        'Fira Sans',
        'Droid Sans',
        'Helvetica Neue',
        sans-serif;
        display: inline-block;
        border-radius: 3px;
        outline: 0;
        overflow: hidden;
        white-space: nowrap;
        animation: pulse 1s ease 0s 1 normal;
        border: none;
        min-width: 1.5ch;
        width: 100%;
        font-weight: 400;
        max-width: 12em;
        text-align: center;
        position: absolute;
        top: 0;
        right: 0;
        bottom: 0;
        left: 0;
        border-bottom: 3px solid hsl(0, 0%, 100%);
        border-bottom-left-radius: 0;
        border-bottom-right-radius: 0;
        cursor: auto;
    }

    @keyframes pulse {
        0% {
            background: rgba(255, 255, 255, .5);
        }

        100% {
            background: rgba(255, 255, 255, 0);
        }
    }

    .name-hidden-span {
        text-align: center;
        line-height: normal;
        min-width: 1.5ch;
        vertical-align: top;
        max-width: 12em;
        display: inline-block;
        border-radius: 30px;
        outline: 0;
        overflow: hidden;
        white-space: nowrap;
        visibility: visible;
        font-size: 100%;
    }
}