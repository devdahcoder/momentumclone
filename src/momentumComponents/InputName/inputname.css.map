{
    "version": 3,
    "mappings": "AAAA,AAAA,qBAAqB,CAAC;EAClB,KAAK,EAAE,IAAI;EACX,WAAW,EAAE,CAAC,CAAC,GAAG,CAAC,GAAG,CAAC,kBAAiB;EACxC,WAAW,EAAE,IAAI;EACjB,UAAU,EAAE,MAAM;EAClB,WAAW,EAAE,GAAG;EAChB,WAAW,EAAE,MAAM;EACnB,SAAS,EAAE,MAAM;EACjB,WAAW,EAAE,MAAM;EACnB,UAAU,EAAE,OAAO;EACnB,MAAM,EAAE,cAAc;EACtB,OAAO,EAAE,WAAW;EACpB,QAAQ,EAAE,QAAQ;CA2DrB;;AAvED,AAcI,qBAdiB,CAcjB,KAAK,CAAC;EACF,KAAK,EAAE,IAAI;EACX,SAAS,EAAE,IAAI;EACf,UAAU,EAAE,GAAG;EACf,cAAc,EAAE,GAAG;EACnB,OAAO,EAAE,YAAY;EACrB,aAAa,EAAE,GAAG;EAClB,OAAO,EAAE,CAAC;EACV,QAAQ,EAAE,MAAM;EAChB,WAAW,EAAE,MAAM;EACnB,SAAS,EAAE,yBAAyB;EACpC,MAAM,EAAE,IAAI;EACZ,WAAW,EAAE,GAAG;EAChB,SAAS,EAAE,IAAI;EACf,UAAU,EAAE,MAAM;EAClB,KAAK,EAAE,IAAI;EACX,QAAQ,EAAE,QAAQ;EAClB,GAAG,EAAE,CAAC;EACN,KAAK,EAAE,CAAC;EACR,MAAM,EAAE,CAAC;EACT,IAAI,EAAE,CAAC;EACP,SAAS,EAAE,KAAK;EAChB,aAAa,EAAE,cAAc;EAC7B,yBAAyB,EAAE,CAAC;EAC5B,0BAA0B,EAAE,CAAC;EAC7B,MAAM,EAAE,IAAI;CACf;;AAED,UAAU,CAAV,KAAU;EACN,EAAE;IACE,UAAU,EAAE,wBAAuB;;EAGvC,IAAI;IACA,UAAU,EAAE,sBAAsB;;;;AAhD9C,AAoDI,qBApDiB,CAoDjB,iBAAiB,CAAC;EACd,KAAK,EAAE,IAAI;EACX,WAAW,EAAE,CAAC,CAAC,GAAG,CAAC,GAAG,CAAC,kBAAiB;EACxC,WAAW,EAAE,IAAI;EACjB,UAAU,EAAE,MAAM;EAClB,WAAW,EAAE,GAAG;EAChB,WAAW,EAAE,MAAM;EACnB,SAAS,EAAE,MAAM;EACjB,UAAU,EAAE,OAAO;EACnB,cAAc,EAAE,GAAG;EACnB,SAAS,EAAE,IAAI;EACf,OAAO,EAAE,YAAY;EACrB,aAAa,EAAE,GAAG;EAClB,OAAO,EAAE,CAAC;EACV,QAAQ,EAAE,MAAM;EAChB,WAAW,EAAE,MAAM;EACnB,SAAS,EAAE,KAAK;EAChB,UAAU,EAAE,MAAM;CACrB",
    "sources": [
        "inputname.scss"
    ],
    "names": [],
    "file": "inputname.css"
}