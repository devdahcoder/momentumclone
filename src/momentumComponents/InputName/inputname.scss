.input-name-container {
    text-align: center;
    font-weight: 500;
    line-height: normal;
    // font-size: 50.5%;
    white-space: nowrap;
    box-sizing: inherit;
    margin: -5px -2px -8px;
    padding: 5px 2px 8px;
    position: relative;

    input {
        color: #fff;
        font-size: 100%;
        background: 0 0;
        vertical-align: top;
        display: inline-block;
        border-radius: 3px;
        outline: 0;
        overflow: hidden;
        white-space: nowrap;
        animation: pulse 1s ease 0s 1 normal;
        border: none;
        font-weight: 500;
        max-width: 12em;
        text-align: center;
        width: 100%;
        position: absolute;
        top: 0;
        right: 0;
        bottom: 0;
        left: 0;
        min-width: 1.5ch;
        border-bottom: 3px solid #fff;
        border-bottom-left-radius: 0;
        border-bottom-right-radius: 0;
        cursor: auto;
    }

    @keyframes pulse {
        0% {
            background: rgba(255, 255, 255, .5);
        }

        100% {
            background: rgba(255, 255, 255, 0);
        }
    }

    .name-hidden-span {
        text-align: center;
        font-weight: 500;
        line-height: normal;
        font-size: 337.5%;
        box-sizing: inherit;
        vertical-align: top;
        max-width: 12em;
        display: inline-block;
        border-radius: 3px;
        outline: 0;
        overflow: hidden;
        white-space: nowrap;
        min-width: 1.5ch;
        visibility: hidden;
    }
}