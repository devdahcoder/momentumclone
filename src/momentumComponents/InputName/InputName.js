import React, { useContext } from 'react'
import "./inputname.css";

//imported utils
import {GreetingContext} from "../../context/GreetingContext";

const InputName = () => {

    const { inputName, setInputName, setName, isEditingName, setIsEditingName } = useContext(GreetingContext);

    const handleName = (e) => {

        setInputName(e.target.value);

    }


    const handleTodoSubmit = (e) => {

        if (e.key === "Enter") {

            e.preventDefault();

            if (inputName.trim() !== "") {

                setName(inputName);

            } 

            else {

                alert("cant happen");

            }

            setInputName("");

            // please leave like this and don't do !isEditingName because it breaks the name input
            // when a new user enters their name.

            setIsEditingName(false);
        }
    }

    

    return (
        <span className="input-name-container">

            <input onKeyPress={handleTodoSubmit} onChange={handleName} value={inputName} spellCheck="false" data-test="name-input" className="name editing pulse" /> 
            
            <span className="name-hidden-span">{inputName}</span>
            
        </span>
    )
}

export default InputName
