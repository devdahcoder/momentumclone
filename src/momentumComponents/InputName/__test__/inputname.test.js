import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect"
import InputName from "../InputName";
import GreetingContextProvider from "../../../context/GreetingContext"

test("should render in the document", () => {

    const { getByTestId } = render(<GreetingContextProvider><InputName /></GreetingContextProvider>);

    const inputNameElement = getByTestId("input-name-container");

    expect(inputNameElement).toHaveClass("input-name-container");

    expect(inputNameElement).toBeInTheDocument();

    expect(inputNameElement).toMatchSnapshot();

})


test("should render name input", () => {

    const { getByTestId } = render(<GreetingContextProvider><InputName /></GreetingContextProvider>);

    const inputName = getByTestId("input-name");

    fireEvent.change(inputName, { target: { value: "John" } });

    expect(inputName).toMatchSnapshot();

    expect(inputName).toBeInTheDocument();

});