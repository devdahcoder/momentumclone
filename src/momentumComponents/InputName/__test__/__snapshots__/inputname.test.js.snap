// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render in the document 1`] = `
<span
  class="input-name-container"
  data-testid="input-name-container"
>
  <input
    class="name editing pulse"
    data-test="name-input"
    data-testid="input-name"
    spellcheck="false"
    value=""
  />
  <span
    class="name-hidden-span"
  />
</span>
`;

exports[`should render name input 1`] = `
<input
  class="name editing pulse"
  data-test="name-input"
  data-testid="input-name"
  spellcheck="false"
  value="John"
/>
`;
